runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -422,16 +428,37 @@ func (q *MetricsViewComparisonToplist) buildMetricsComparisonTopListSQL(mv *runt
 		twiceTheLimitClause = fmt.Sprintf(" LIMIT %d", q.Limit*2)
 	}
 
+	baseLimitClause := ""
+	comparisonLimitClause := ""
+
+	joinType := "FULL"
+	if !q.Exact {
+		approximationLimit := q.Limit
+		if q.Limit < 100 {
+			approximationLimit = 100
+		}
```

Comment:
1. Having a high default should only for deltas, right? I don't see how it would change things for base or comparison values
2. How does increasing it impact performance? I wonder if we could increase it to e.g. 500 or 1k for deltas without a performance hit

Reply:
2. See this benchmarks, there's an impact but it will depend on dataset size and cardinality, it's mild with the `spending` dataset:
```
BenchmarkMetricsViewsComparison_high_cardinality_compare_spending-10                                 	      12	  91451069 ns/op	  242984 B/op	    8010 allocs/op
BenchmarkMetricsViewsComparison_high_cardinality_compare_spending_approximate_limit_250-10           	      50	  24072948 ns/op	  242431 B/op	    7998 allocs/op
BenchmarkMetricsViewsComparison_delta_high_cardinality_compare_spending_approximate_limit_500-10     	      37	  29629658 ns/op	  458121 B/op	   15155 allocs/op
BenchmarkMetricsViewsComparison_delta_high_cardinality_compare_spending_approximate_limit_1250-10    	      37	  30443241 ns/op	 1109369 B/op	   36573 allocs/op
```